### PR TITLE
Fixes #1974 Upgrade to akka-2.4.2

### LINKED ISSUE #1974: Upgrade to akka-stream-experimental_2.11 2.0.3
See - http://akka.io/news/2016/02/17/akka-2.4.2-released.html. This includes akka-stream and Http 2.0.3.

### PR DESCRIPTION
**Pull request by kkasravi**
For review right now. Many changes coming.

### CODE REVIEW

**Comment by manuzhang** [comment]
build failed since akka 2.4.2 doesn't support scala 2.10

File: core/src/main/scala/io/gearpump/transport/netty/Server.scala [hunk 1]
```diff
@@ -116,7 +117,10 @@ object Server {
 
     def translateToActorRef(sessionId : Int): ActorRef = {
       if(!taskIdtoActorRef.contains(sessionId)){
-        val actorRef = context.system.actorFor(s"/session#$sessionId")
+        import scala.concurrent.duration._
+        implicit val timeout = Timeout(5 seconds)
+        val actorRef = Await.result(context.system.actorSelection(s"/session#$sessionId").resolveOne(), timeout.duration)
```
Comment (huafengw) [inline core/src/main/scala/io/gearpump/transport/netty/Server.scala]: We'd better avoid using Await and find another solution here.

**Comment by manuzhang** [comment]
could you create a new pr against akka-stream branch ?

**Comment by clockfly** [comment]
@kkasravi 

Should we create a akka2.4 branch? Since akka2.4 means a lot of incompatible things, like JDK, like akka-stream, like scala2.10 support.

**Comment by kkasravi** [comment]
@clockfly I think maybe they can be handled separately. @helena also opened a similar issue #1988 

I've changed this Build.scala to use akka 2.3.14 and akka-stream-experimental 2.0.3. 
Note that are some differences between akka-stream which is included in 2.4.2 and akka-stream-experimental 2.0.3.

File: experiments/akkastream/src/main/scala/akka/stream/ModuleGraph.scala [hunk 2]
```diff
@@ -146,6 +146,37 @@ object ModuleGraph {
     private def publishers: mutable.Map[OutPort, (OutPort, Module)] = publishersStack.head
     private def currentLayout: Module = moduleStack.head
 
+    // Enters a copied module and establishes a scope that prevents internals to leak out and interfere with copies
```
Comment (clockfly) [inline experiments/akkastream/src/main/scala/akka/stream/ModuleGraph.scala]: Please don't change the method so that we don't generate unnecessary diff.

**Comment by helena** [comment]
I'm working on a crossbuild solution for the upgrade, will go toward https://github.com/gearpump/gearpump/issues/1988. Work started, sidetracked this week for an O'Reilly webcast I just did, I should be able to PR Friday/Saturday. Hopefully we can all collaborate on that ;)

**Comment by kkasravi** [comment]
@helena thanks. Send us a link to your webcase :)
We will be submitting a PR to the akka team based on 2.4.2 for a few classes we've modified as part of the akka-stream-experimental integration. I'll rebase from #1988 once we've integrated your changes.

**Comment by clockfly** [comment]
@helena 

```
I'm working on a crossbuild solution for the upgrade
```

Do you mean cross build on scala2.11 and scala2.12?

**Comment by helena** [comment]
@clockfly No, 2.11.7 and backward compat for 2.3.14 deps to build. I don't think 2.12 would resolve anything and is not GA, only 2.11.8.

**Comment by helena** [comment]
@kkasravi I'm not sure when o'reilly will release the vid of our webcast but it was http://www.oreilly.com/pub/e/3652 and our slides at least are http://www.slideshare.net/helenaedelson/fast-and-simplified-streaming-adhoc-and-batch-analytics-with-filodb-and-spark-streaming :)

**Comment by kkasravi** [comment]
@helena really great deck on filodb advantages compared to cassandra, the gearpump-external-filodb has us interested :)
For #1988  you may want to start with the Build.scala I've been working with which uses 2.4.2.
https://github.com/gearpump/gearpump/pull/1978/files/d18d4749f0d561d8dcaa786d94ef9a3bfc279662#diff-215113124f1de02f228327ba7abb45f7
